services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/extension/env/EnvCleanupExtension.groovy

```diff
@@ -0,0 +1,155 @@
+
+    @Autowired
+    TopologyDefinition topology
+
+    @Autowired
+    NorthboundService northbound
+
+    @Autowired
+    Database database
+
+    @Autowired
+    IslUtils islUtils
+
+    @Autowired
+    LockKeeperService lockKeeper
+
+    @Value('${spring.profiles.active}')
+    String profile
+
+    @Override
+    void start() {
+        SpringContextExtension.addListener(this)
+    }
+
+    def deleteAllFlows() {
+        log.info("Deleting all flows")
+        northbound.deleteAllFlows()
+    }
+
+    def unsetLinkMaintenance(List<IslInfoData> links) {
+        def maintenanceLinks = northbound.getAllLinks().findAll { it.underMaintenance }
+        if (maintenanceLinks) {
+            log.info("Unset maintenance mode for affected links: $maintenanceLinks")
+            maintenanceLinks.each {
+                northbound.setLinkMaintenance(islUtils.getLinkUnderMaintenance(it, false, false))
+            }
+        }
+    }
+
+    def reviveFailedLinks(List<IslInfoData> links) {
+        def failedLinks = links.find { it.state == IslChangeType.FAILED }
+        if (failedLinks) {
+            log.info("Bring ports up for failed links: $failedLinks")
+            failedLinks.each {
+                try {
+                    northbound.portUp(it.source.switchId, it.source.portNo)
+                    northbound.portUp(it.destination.switchId, it.destination.portNo)
+                } catch (Throwable t) {
+                    // Switch may be disconnected. Ignore for now, healthcheck test will report problems if any.
+                    // This situation may require manual intervention
+                }
+            }
+        }
+    }
+
+    def deleteLinkProps() {
+        def linkProps = northbound.getAllLinkProps()
+        if (linkProps) {
+            log.info("Deleting all link props")
+            northbound.deleteLinkProps(linkProps)
+        }
+    }
+
+    def resetCosts() {
+        log.info("Resetting all link costs")
+        database.resetCosts()
+    }
+
+    def resetBandwidth(List<IslInfoData> links) {
+        def topoIsls = topology.isls.collectMany { [it, islUtils.reverseIsl(it)] }
+        links.each { link ->
+            if (link.maxBandwidth != link.availableBandwidth || link.maxBandwidth != link.speed) {
+                def isl = topoIsls.find {
+                    it.srcSwitch.dpId == link.source.switchId && it.srcPort == link.source.portNo &&
+                            it.dstSwitch.dpId == link.destination.switchId && it.dstPort == link.destination.portNo
+                }
+                assert isl
+                log.info("Resetting available bandwidth on ISL $isl")
+                database.revertIslBandwidth(isl)
+            }
+        }
+    }
+
+    def unsetSwitchMaintenance() {
+        def maintenanceSwitches = northbound.getAllSwitches().findAll { it.underMaintenance }
+        if (maintenanceSwitches) {
+            log.info("Unset maintenance mode from all affected switches")
```

Comment:
Isn't logical to do the same for switches as for links? Or there is another logic and we log only that what is related to links?

Reply:
done